src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,61 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String CHECK_SUPPLY = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] dataFromFile = readFile(fromFileName);
+        String[] calculatedStatistic = calculateStatistic(dataFromFile);
+        writeFile(calculatedStatistic, toFileName);
+    }
+
+    private String[] readFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(" ");
+                value = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file", e);
+        }
+        return stringBuilder.toString().split(" ");
+    }
+
+    private void writeFile(String[] calculatedStatistic, String toFileName) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
+            for (String data : calculatedStatistic) {
+                bufferedWriter.write(data);
+                bufferedWriter.write(System.lineSeparator());
+                bufferedWriter.flush();
```

Comment:
It is not needed `flush()` data in the loop, it will be very slow down the program execution. 
read about `flush()` e.g. here
https://stackoverflow.com/questions/908168/when-to-flush-a-bufferedwriter
For saving to file optimization, we can create a result concatenated String (remember do not concatenate strings in the loop) and then save the result string to the file just one time.